native/src/main/java/io/ballerina/stdlib/log/MaskedStringBuilder.java

```diff
@@ -0,0 +1,578 @@
+     */
+    private void appendEscapedString(String input) {
+        if (input == null) {
+            this.stringBuilder.append("null");
+            return;
+        }
+
+        if (!needsEscaping(input)) {
+            this.stringBuilder.append(input);
+            return;
+        }
+
+        for (int i = 0; i < input.length(); i++) {
+            char c = input.charAt(i);
+            switch (c) {
+                case '"' -> this.stringBuilder.append(QUOTE_ESCAPE);
+                case '\\' -> this.stringBuilder.append(BACKSLASH_ESCAPE);
+                case '\b' -> this.stringBuilder.append(BACKSPACE_ESCAPE);
+                case '\f' -> this.stringBuilder.append(FORM_FEED_ESCAPE);
+                case '\n' -> this.stringBuilder.append(NEWLINE_ESCAPE);
+                case '\r' -> this.stringBuilder.append(CARRIAGE_RETURN_ESCAPE);
+                case '\t' -> this.stringBuilder.append(TAB_ESCAPE);
+                default -> {
+                    // Escape ASCII control characters (0x00-0x1F) and DEL character (0x7F)
+                    if (c < ASCII_CONTROL_CHAR_LIMIT || c == ASCII_DEL_CHAR) {
+                        this.stringBuilder.append("\\u00");
+                        this.stringBuilder.append(HEX_CHARS[(c >>> 4) & 0xF]);
+                        this.stringBuilder.append(HEX_CHARS[c & 0xF]);
+                    } else {
+                        this.stringBuilder.append(c);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Get cached field annotations for better performance.
+     * Implements simple cache eviction when cache grows too large.
+     */
+    private Map<String, BMap<?, ?>> getCachedFieldAnnotations(RecordType recordType) {
+        Map<String, BMap<?, ?>> cached = ANNOTATION_CACHE.get(recordType);
+        if (cached != null) {
+            return cached;
+        }
+
+        // Implement simple cache size management
+        if (ANNOTATION_CACHE.size() >= MAX_CACHE_SIZE) {
+            // Clear half the cache when it gets too large (simple eviction strategy)
+            ANNOTATION_CACHE.entrySet().removeIf(entry -> System.identityHashCode(entry.getKey()) % 2 == 0);
+        }
+
+        Map<String, BMap<?, ?>> annotations = extractFieldAnnotations(recordType);
+        ANNOTATION_CACHE.put(recordType, annotations);
+        return annotations;
+    }
+
+    private String processTableValue(BTable<?, ?> tableValue) {
+        Collection<?> values = tableValue.values();
+        if (values.isEmpty()) {
+            return "[]";
+        }
+
+        int startPos = this.stringBuilder.length();
+        this.stringBuilder.append('[');
+
+        boolean first = true;
+        for (Object row : values) {
+            if (!first) {
+                this.stringBuilder.append(',');
+            }
+            String elementString = buildInternal(row);
+            appendValueToArray(elementString, row);
+            first = false;
```

Comment:
```suggestion
                this.stringBuilder.append(',');
                first = false;
            }
            String elementString = buildInternal(row);
            appendValueToArray(elementString, row);
```

Reply:
Why this change? I set `first` to true initially and after adding the very first element, I set it to `false` so that the next fields will be added after a comma